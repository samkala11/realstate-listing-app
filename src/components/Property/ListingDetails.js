import React from 'react';
 
const ListingDetails = ({ listingDate, mlsId }) => {
    return(
        <div className="listing-details">
            { listingDate && <p className="date-listed"> Date listed <span className="date-display"> 
                {listingDate.slice(5,10).split('-').join('/')}/{listingDate.slice(0,4)} </span> </p> }
            <p className="mls-id"> mlsId <span className="mls-id-display"> {mlsId} </span> </p>
        </div>
    )
}

export default ListingDetails;