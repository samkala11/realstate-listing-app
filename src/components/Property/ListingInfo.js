import React from 'react';
import { formulatePrice } from '../../utils/price';
 
const ListingInfo = ({ price, bedrooms, bathsFull, bathsHalf, stories }) => {
    return(
        <div className="listing-info">
            <p  data-testid = "listing-price" className="listing-price"> ${formulatePrice(price)} </p>
            { bedrooms && <p className="bedrooms-number">  {bedrooms} {bedrooms > 1 ? 'bds' : 'bd' } | </p> } 
            <p className="bath-number" > {bathsFull + bathsHalf} ba | </p>
            <p className="stories-number"> {stories} stories </p>
        </div>
    )
}

export default ListingInfo;