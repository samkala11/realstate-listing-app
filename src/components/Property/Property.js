import React from 'react';
import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ListingInfo from './ListingInfo';
import ListingDetails from './ListingDetails';

const Property = ({mlsId, photosArray, price, bedrooms, bathsFull,
     bathsHalf, stories, saveProperty, listingDate, saved, id}) => {

    return( <div data-testid="property-listing-container" className="property-listing">
                
        <button
            onClick={() => saveProperty(mlsId, saved, id)} 
            id="save-button-id" className="save-button"
        >
            { saved ? <FavoriteIcon/> : <FavoriteBorderIcon/>  }
        </button>
            
        { photosArray && <img className="property-image" alt="listing" src={photosArray[0]}/> }

        <ListingInfo
            price = {price}
            bedrooms = {bedrooms}
            bathsFull = {bathsFull}
            bathsHalf = {bathsHalf}
            stories = {stories}     
        />

        <ListingDetails 
            listingDate = {listingDate}
            mlsId = {mlsId} 
        />
   
    </div>
    )
}

export default Property;