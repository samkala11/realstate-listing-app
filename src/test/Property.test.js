import { render, screen, cleanup } from '@testing-library/react';
import Property from '../components/Property/Property';
import renderer from 'react-test-renderer';

afterEach(cleanup);


test('Renders Property page successfully', () => {

    render(<Property  
            price={937477}
            mlsId={5990355}
            bedrooms = {3}
            bathsFull = {4}
            bathsHalf={2}
            stories={2}
            listingDate={'1994-10-25T13:58:17.284009Z'}
            saved={true}
        />);

    const propertyListingContainer = screen.getByTestId('property-listing-container');
    expect(propertyListingContainer).toBeInTheDocument();
});


test('Shows property info correctly', () => {

    render(<Property  
        price={937477}
        mlsId={5990355}
        bedrooms = {3}
        bathsFull = {4}
        bathsHalf={2}
        stories={2}
        listingDate={'1994-10-25T13:58:17.284009Z'}
        saved={true}
    />);

    const listingPrice = screen.getByTestId('listing-price');
    expect(listingPrice).toHaveTextContent('$937,477');
    
    const bedroomsNumber = screen.getByTestId('bedrooms-number');
    expect( bedroomsNumber).toHaveTextContent('3');
    
    const bathNumber = screen.getByTestId('bath-number');
    expect( bathNumber).toHaveTextContent('6');
    
    const storiesNumber = screen.getByTestId('stories-number');
    expect( storiesNumber).toHaveTextContent('2');

});



test('matches Property snapshot with given props', () => {
    const tree = renderer.create(<Property 
            price={937477}
            mlsId={5990355}
            bedrooms = {3}
            bathsFull = {4}
            bathsHalf={2}
            stories={2}
            listingDate={'1994-10-25T13:58:17.284009Z'}
            saved={true}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})