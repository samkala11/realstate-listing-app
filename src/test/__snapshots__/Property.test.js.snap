// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Property snapshot with given props 1`] = `
<div
  className="property-listing"
  data-testid="property-listing-container"
>
  <button
    className="save-button"
    id="save-button-id"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
      />
    </svg>
  </button>
  <div
    className="listing-info"
  >
    <p
      className="listing-price"
      data-testid="listing-price"
    >
       $
      937,477
       
    </p>
    <p
      className="bedrooms-number"
    >
        
      3
       
      bds
       | 
    </p>
    <p
      className="bath-number"
    >
       
      6
       ba | 
    </p>
    <p
      className="stories-number"
    >
       
      2
       stories 
    </p>
  </div>
  <div
    className="listing-details"
  >
    <p
      className="date-listed"
    >
       Date listed 
      <span
        className="date-display"
      >
        10/25
        /
        1994
         
      </span>
       
    </p>
    <p
      className="mls-id"
    >
       mlsId 
      <span
        className="mls-id-display"
      >
         
        5990355
         
      </span>
       
    </p>
  </div>
</div>
`;
